### PR TITLE
Google schema guided dialog dataset with API calls and responses

### PR DESCRIPTION
**Pull request by chinnadhurai**
**Patch description**

Adds Google's [schema guided dialog dataset](https://github.com/google-research-datasets/dstc8-schema-guided-dialogue)

**Testing steps**

python tests/datatests/test_new_tasks.py

> 10:42:41 INFO | creating task(s): google_sgd:DefaultTeacher
10:42:41 WARN | Google SGD is a beta dataset, and format may significantly change.
10:43:50 INFO | Loaded 16142 episodes with a total of 215128 examples
10:43:51 INFO | creating task(s): google_sgd:Text2API2TextTeacher
10:44:59 INFO | Loaded 16142 episodes with a total of 215128 examples
10:44:59 INFO | creating task(s): google_sgd:Text2TextTeacher
10:45:36 INFO | Loaded 16142 episodes with a total of 164982 examples
10:45:36 INFO | creating task(s): google_sgd:DefaultTeacher
10:46:44 INFO | Loaded 16142 episodes with a total of 215128 examples
10:46:45 INFO | creating task(s): google_sgd:Text2API2TextTeacher
10:47:54 INFO | Loaded 16142 episodes with a total of 215128 examples
10:47:54 INFO | creating task(s): google_sgd:Text2TextTeacher
10:48:30 INFO | Loaded 16142 episodes with a total of 164982 examples
10:48:30 INFO | creating task(s): google_sgd:DefaultTeacher
10:49:40 INFO | Loaded 16142 episodes with a total of 215128 examples
10:49:40 INFO | creating task(s): google_sgd:Text2API2TextTeacher
10:50:50 INFO | Loaded 16142 episodes with a total of 215128 examples
10:50:50 INFO | creating task(s): google_sgd:Text2TextTeacher
10:51:28 INFO | Loaded 16142 episodes with a total of 164982 examples
.
----------------------------------------------------------------------
Ran 1 test in 527.828s

OK

**Sample Dialogs**

python examples/display_data.py -t google_sgd:Text2TextTeacher

>User: I am feeling hungry so I would like to find a place to eat.
Sys:  Do you have a specific which you want the eating place to be located at?
User: I would like for it to be in San Jose.
Sys: Is there a specific cuisine type you enjoy, such as Mexican, Italian or something else?
User: I usually like eating the American type of food.
Sys: I see that at 71 Saint Peter there is a good restaurant which is in San Jose.
User: Can you give me the address of this restaurant.
Sys: If you want to go to this restaurant you can find it at 71 North San Pedro Street.
User: Can you give me the phone number that I can contact them with?
Sys: If you want to phone them you can at 408-971-8523.
User: Is there some other restaurant which you can suggest?
Sys: How would you like Bazille restaurant which is situated in San Jose.
User: Do you have another restaurant matching my needs? For example a restaurant which is economical and is located in Palo Alto.
Sys: I see that 7 restaurants suit to what you requested. Bird Dog seems as a good restaurant and is located in Palo Alto.
User: Alright, that seems good. I would like to make a booking at this restaurant.
Sys: For which time do you want the booking to be?
User: I will be eating there at 11:30 am so make it for then.
Sys: Can you please confirm that you want to book a table for 2 at 11:30 am at the Bird Dog restaurant in Palo Alto for today.
User: That suits me well. Can you tell me if they feature live music?
Sys: Your booking has been made without errors, but unfortunately they do not have live music.

### CODE REVIEW

File: parlai/core/build_data.py [hunk 1]
```diff
@@ -153,6 +153,7 @@ def download(url, path, fname, redownload=False, num_retries=5):
     If ``redownload`` is set to false, then will not download tar file again if it is
     present (default ``False``).
     """
+    make_dir(path)
```
Comment (stephenroller) [inline parlai/core/build_data.py:156]: This shouldn't make it in.

Comment (chinnadhurai) [reply parlai/core/build_data.py]: done

**Comment by chinnadhurai** [comment]
python tests/datatests/test_new_tasks.py

10:42:41 INFO | creating task(s): google_sgd:DefaultTeacher
10:42:41 WARN | Google SGD is a beta dataset, and format may significantly change.
10:43:50 INFO | Loaded 16142 episodes with a total of 215128 examples
10:43:51 INFO | creating task(s): google_sgd:Text2API2TextTeacher
10:44:59 INFO | Loaded 16142 episodes with a total of 215128 examples
10:44:59 INFO | creating task(s): google_sgd:Text2TextTeacher
10:45:36 INFO | Loaded 16142 episodes with a total of 164982 examples
10:45:36 INFO | creating task(s): google_sgd:DefaultTeacher
10:46:44 INFO | Loaded 16142 episodes with a total of 215128 examples
10:46:45 INFO | creating task(s): google_sgd:Text2API2TextTeacher
10:47:54 INFO | Loaded 16142 episodes with a total of 215128 examples
10:47:54 INFO | creating task(s): google_sgd:Text2TextTeacher
10:48:30 INFO | Loaded 16142 episodes with a total of 164982 examples
10:48:30 INFO | creating task(s): google_sgd:DefaultTeacher
10:49:40 INFO | Loaded 16142 episodes with a total of 215128 examples
10:49:40 INFO | creating task(s): google_sgd:Text2API2TextTeacher
10:50:50 INFO | Loaded 16142 episodes with a total of 215128 examples
10:50:50 INFO | creating task(s): google_sgd:Text2TextTeacher
10:51:28 INFO | Loaded 16142 episodes with a total of 164982 examples
.
----------------------------------------------------------------------
Ran 1 test in 527.828s

OK

File: parlai/tasks/google_sgd/agents.py [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+                        'type': 'apiresp',
+                        'slots': api_results,
+                    }, False
+                else:
+                    assert (
+                        api_call and api_results
+                    ), "API call without API results! Check Dataset!"
+
+
+class Text2TextTeacher(Text2API2TextTeacher):
```
Comment (stephenroller) [inline parlai/tasks/google_sgd/agents.py:168]:
```suggestion
class Text2TextTeacher(Text2API2TextTeacher):
    """
    Text-only teacher (with no API calls or slots)
    """
```

**Pull request merged**